corehq/apps/products/forms.py

```diff
@@ -42,9 +49,16 @@ def __init__(self, product, *args, **kwargs):
         # make sure to select default program if
         # this is a new product
         if not product._id:
-            self.initial['program_id'] = Program.default_for_domain(
-                self.product.domain
-            )._id
+            self.initial['program_id'] = Program.default_for_domain(self.product.domain)._id
```

Comment:
That program should be created when [`make_domain_commtrack`](https://github.com/dimagi/commcare-hq/blob/a54a145f000271a7b195a5158830ae660c6bc8b0/corehq/apps/domain/models.py#L1063) gets called, which happens when you enable the feature flag, or during tests [here](https://github.com/dimagi/commcare-hq/blob/a54a145f000271a7b195a5158830ae660c6bc8b0/corehq/apps/commtrack/tests/util.py#L53).  If your domain had commtrack enabled by some other means, you could disable, then re-enable the feature flag to get that called again.